src/main/java/org/scijava/ui/swing/script/autocompletion/JythonAutoCompletion.java

```diff
@@ -65,7 +66,26 @@ public Import(final String packageName, final String[] parts, final int lineNumb
 			this(packageName + "." + parts[0], 3 == parts.length ? parts[2] : null, lineNumber);
 		}
 	}
-	
+
+	static public final String findClassAliasOfVariable(final String variable, String inputText) {
+		final String[] lines = inputText.split("\n");
+		for (int i = 0; i < lines.length; ++i) {
+			final String line = lines[i];
+			final Matcher matcher = variableDeclarationPattern.matcher(line);
+			if (matcher.find()) {
+				// a line containing a variable declaration
+//				System.out.println("Queried variable: " + variable);
+//				System.out.println("Hit: line #" + i + ": " + line);
+//				System.out.println("Matcher g1: " + matcher.group(1));
+//				System.out.println("Matcher g2: " + matcher.group(2));
+				if (variable.equals(matcher.group(1))) {
```

Comment:
This approach is dangerous in that it doesn't consider the scope of the variable. I would do it differently, by parsing the code like jython's interpreter with a ParserFacade, see e.g. https://github.com/acardona/scripts/blob/master/python/imagej/jython/parse_code.py

Reply:
Good point. Are you able to work on this? To me the priority would be to support he return type of the static method

Reply:
Yes, proper jython code parsing is something I've been working on but haven't committed yet. Will lead to:
1. autocompletion for variable names within scope, in addition to possible java class imports;
2. autocompletion and auto-import for python modules included in the jython jar or discoverable from its PYTHONPATH-equivalent;
3. type resolution for variables within scope, in order to autocomplete their methods. Won't always work (e.g. when an object derives from a call on a function argument), but it will work in many cases.

Reply:
@acardona, Great! On a second thought I will hold for now, as you are likely to revamp a bunch of things. But do let me know if there is something you want me too look into. I won't work on this until I hear from you (so feel free to merge), but do let me know if you want me to look into something. I will post below a summary of what this (snowballed) PR now has.